build.gradle.kts

```diff
@@ -1,226 +1,193 @@
-import com.adarshr.gradle.testlogger.theme.ThemeType
-import org.jetbrains.intellij.tasks.RunPluginVerifierTask.VerificationReportsFormats.*
-import org.jetbrains.intellij.tasks.RunPluginVerifierTask.FailureLevel.*
-
-fun properties(key: String) = providers.gradleProperty(key)
-fun environment(key: String) = providers.environmentVariable(key)
-fun prop(name: String): String {
-    return properties(name).get()
-}
+import org.jetbrains.changelog.Changelog
+import org.jetbrains.changelog.markdownToHTML
+import org.jetbrains.intellij.platform.gradle.TestFrameworkType
+
 plugins {
-    java // Java support
+    id("java") // Java support
     alias(libs.plugins.kotlin) // Kotlin support
-    alias(libs.plugins.gradleIntelliJPlugin) // Gradle IntelliJ Plugin
+    alias(libs.plugins.intelliJPlatform) // IntelliJ Platform Gradle Plugin
     alias(libs.plugins.changelog) // Gradle Changelog Plugin
-    alias(libs.plugins.testLogger) // Nice test logs
+    alias(libs.plugins.qodana) // Gradle Qodana Plugin
     alias(libs.plugins.kover) // Gradle Kover Plugin
-    jacoco // Code coverage
 }
 
-group = prop("pluginGroup")
-version = prop("pluginVersion")
+group = providers.gradleProperty("pluginGroup").get()
+version = providers.gradleProperty("pluginVersion").get()
 
-val lsp4jVersion = prop("lsp4jVersion")
-val flexmarkVersion = prop("flexmarkVersion")
+// Set the JVM language level used to build the project.
+kotlin {
+    jvmToolchain(17)
+}
 
 // Configure project's dependencies
 repositories {
-    mavenLocal()
     mavenCentral()
+
+    // IntelliJ Platform Gradle Plugin Repositories Extension - read more: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-repositories-extension.html
+    intellijPlatform {
+        defaultRepositories()
+    }
+
+    mavenLocal()
     maven { url = uri("https://repository.jboss.org/nexus/content/repositories/snapshots") }
     maven { url = uri("https://repository.jboss.org/nexus/content/groups/public") }
     maven { url = uri("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
```

Comment:
this line can be safely removed because it's actually included in the default intellij platform repository declaration above.